lego-mindstorms-pieces.py

```diff
@@ -221,49 +268,67 @@ def order(shop=None, browser=None, lego_set=None, order_list=None, username=None
-        print("- {qty}x #{pn} ".format(qty=quantity, pn=part_no), end='')
 
+        counter += 1
+
+        print("- [{counter}/{total_elements}] {qty}x #{pn} ".format(qty=quantity, pn=part_no,counter=counter,total_elements=total_elements), end='')
+
+        element_field = wait.until(EC.element_to_be_clickable((By.ID, 'element-filter')))
         element_field.clear()
         element_field.send_keys(part_no)
         element_field.send_keys(Keys.RETURN)
         sleep(.3)  # seconds
 
         try:
             add_button = browser.find_element_by_css_selector('.element-details + button')
-            add_button.click()
-            sleep(.2)  # seconds
+            if add_button.is_enabled():
+              add_button.click()
+              sleep(.2)  # seconds
+            else:
+              counter_outOfStock += 1
+              print("NOTE: item out of stock. ")
+              continue
+
+            print("Found ", end='')
+
+            # set the value for item's quantity drop-down menu
+            amount_select = browser.find_elements_by_css_selector('.bag-item select')[-1]
+            Select(amount_select).select_by_visible_text(quantity)
+
+            # ensure the value is correct
+            selected = Select(amount_select).first_selected_option
+
+            if quantity != selected.text:
+              print("WARNING: Could not select desired quantity. {} != {}".format(quantity, selected.text))
+            else:
+              counter_found +=1
+              print()
+
         except NoSuchElementException:
             print("OOOPS! No LEGO part with that number found in set #{set}. :-(".format(set=lego_set))
+            counter_notInSet += 1
             continue
 
-        try:
-            warn_msg = browser.find_element_by_css_selector('.alert-warning .sold-out-info')
-            if warn_msg.is_displayed():
-                print("NOTE: item out of stock. ", end='')
-                add_anyway = browser.find_element_by_css_selector('.alert-warning + .clearfix button')
-                add_anyway.click()
-        except NoSuchElementException:
-            pass
-
-        amount_select = browser.find_elements_by_css_selector('.bag-item select')[-1]
-        amount_select.send_keys(quantity)
-        amount_select.send_keys(Keys.TAB)
-
-        selected = Select(amount_select).first_selected_option
-        if quantity != selected.text:
-            print("WARNING: Could not select desired quantity. {} != {}".format(quantity, selected.text))
-        else:
-            print()
 
     browser.execute_script("window.scroll(0, 0);")
     print()
     print("We're done. You can finalize your order now. Thanks for watching!")
+    print()
+    print("Stats :")
+    print("Elements found          : {s}/{total_elements}".format(s=counter_found,total_elements=total_elements))
+    print("Elements not in set     : {s}".format(s=counter_notInSet))
+    print("Elements out of stock   : {s}".format(s=counter_outOfStock))
```

Comment:
In general, no spaces before punctuations in English. And, what would you think if we got rid of the aligning problem by placing the numbers in front, hmm? Would that be okay? Kind of ...
```
Statistics:
- xx elements found.
- yy elements not in set.
- zz elements out of stock.
```
No worries about alignment then.